### PR TITLE
chore: restrict function visibility where overly broad

### PR DESCRIPTION
**Pull request by TomAFrench**
# Related issue(s)

Addresses #919 for all crates other than `noirc_evaluator` and `noirc_frontend`

# Description

## Summary of changes

I've added the `unreachable_pub` to several crates and restricted function visibility to the minimum necessary.

## Test additions / changes

# Checklist

- [x] I have tested the changes locally.
- [x ] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` with default settings.
- [ ] I have [linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) this PR to the issue(s) that it resolves.
- [x] I have reviewed the changes on GitHub, line by line.
- [x] I have ensured all changes are covered in the description.
- [ ] This PR requires documentation updates when merged.

# Additional context

### CODE REVIEW

File: crates/nargo/src/toml.rs [hunk 1]
```diff
@@ -5,15 +5,16 @@ use std::path::Path;
 use crate::errors::CliError;
 
 #[derive(Debug, Deserialize, Clone)]
-pub struct Config {
-    pub package: Package,
-    pub dependencies: BTreeMap<String, Dependency>,
+pub(crate) struct Config {
+    #[allow(dead_code)]
```
Comment (kevaundray) [inline crates/nargo/src/toml.rs:9]: why is the dead_code lint needed?

**Review by kevaundray — APPROVED** [review]
LGTM - one question re dead code

File: crates/nargo/src/toml.rs (hunk 1 above)
Comment (TomAFrench) [reply crates/nargo/src/toml.rs:9]: We never actually read the `package` field anywhere. This isn't currently picked up as it's listed as public (despite not being in the crate interface) but when we restrict its visibility to the crate then rust can see that we don't use it.

Comment (kevaundray) [reply crates/nargo/src/toml.rs:9]: Ahh , okay I think thats a separate issue that we need to look into in the future, re where are the dead_code lints

**Pull request merged**